nodejs/eks/cluster.ts

```diff
@@ -102,6 +103,49 @@ function createOrGetInstanceProfile(name: string, parent: pulumi.ComponentResour
   return instanceProfile;
 }
 
+function createEntityRole(name: string, parent: pulumi.Resource): pulumi.Output<aws.iam.Role> {
+    const iamRole = new aws.iam.Role(`${name}-eksClusterCreatorRole`, {
+        assumeRolePolicy: aws.getCallerIdentity().then(id => `{
+            "Version": "2012-10-17",
+            "Statement": [
+                {
+                "Effect": "Allow",
+                "Principal": {
+                    "AWS": "arn:aws:iam::${id.accountId}:root"
+                },
+                "Action": "sts:AssumeRole"
+                }
+            ]
+            }`,
+        ),
+        description: `Admin access to eks-${name}`,
+      }, { parent: parent },
+    );
+
+    const rolePolicy = new aws.iam.RolePolicy(`${name}-eksClusterCreatorPolicy`, {
+        role: iamRole,
+        policy: {
+          Version: "2012-10-17",
+          Statement: [
+            {
+              Effect: "Allow",
+              Action: "eks:*",
+              Resource: "*",
+            },
+            {
+              Effect: "Allow",
+              Action: "iam:PassRole",
+              Resource: "*",
+            },
```

Comment:
Are these required permissions documented anywhere in AWS docs we could link to here?

Reply:
The `eks:*` is needed to create, read, update, delete the EKS cluster. The `iam:Passrole` is to give the EKS cluster the `serviceRole` that gets created in the class. https://github.com/pulumi/pulumi-eks/pull/205/files#diff-0dfaec1e9f3a8568a8e2516febe83a69R173

https://docs.aws.amazon.com/eks/latest/userguide/service_IAM_role.html